### PR TITLE
Add initial connection parameter to WebSocketTransport

### PR DESCRIPTION
**Pull request by winstondu**
Our proposed solution to #1457

### CODE REVIEW

**Comment by designatednerd** [comment]
@winstondu Thanks for the PR! Are you assuming that the developer would just need to call `updateHeaderValues` or `updateConnectingPayload`, which both call `reconnectWebSocket` under the hood?

File: Sources/ApolloWebSocket/WebSocketTransport.swift [hunk 1]
```diff
@@ -112,6 +113,7 @@ public class WebSocketTransport {
               reconnect: Bool = true,
               reconnectionInterval: TimeInterval = 0.5,
               allowSendingDuplicates: Bool = true,
+              connectOnInit: Bool = true,
```
Comment (designatednerd) [inline Sources/ApolloWebSocket/WebSocketTransport.swift:116]: Can you please add documentation for this parameter?

**Comment by designatednerd** [comment]
Test failures seem to be unrelated, I'm not sure what the hell is going on with circle right now

**Comment by designatednerd** [comment]
Are you able to build this locally? If so, would you mind rebasing back onto `main` in the hopes that it'll kick Circle hard enough for this to actually work? I don't see anything in here that would reasonably be causing the type of failure I'm seeing in circle without it failing on every other branch

**Comment by winstondu** [comment]
@designatednerd , done.

File: Sources/ApolloWebSocket/WebSocketTransport.swift (hunk 1 above)
Comment (designatednerd) [reply Sources/ApolloWebSocket/WebSocketTransport.swift:116]: ^can you update this documentation pls?

**Review by designatednerd — COMMENTED** [review]
OK good other than the docs

**Comment by designatednerd** [comment]
LOL CI is such a black box.

**Comment by winstondu** [comment]
> OK good other than the docs

Documentation updated.

**Review by designatednerd — APPROVED** [review]
Approved pending CI

**Pull request merged**